### ISSUE TITLE
Cannot set custom setting with MONGODB_SETTINGS

### ISSUE THREAD

**Issue by ntcong** [MongoEngine/flask-mongoengine#250]
Since fce79d8adaac528bca4c7301f4a5e36a40d7e0e4, only a few settings from conn_settings will get passed down to resolved. 

This results in a removal of TZ_AWARE and possibly some other mongoengine setting.

**Comment by ntcong** [comment]
This also raises a problem with test coverage. I saw some test with tz_aware before and then removed.

How should I address this issue? I can do a PR but I want to know how the dev want to address the problem: either we pass all the settings to mongoengine like 0.7.x, or I can find all the popular setting and set each of them from there.

**Comment by lafrech** [comment]
@ntcong, thanks for your help.

From [here](https://github.com/MongoEngine/flask-mongoengine/issues/247#issuecomment-243348557) and [here](https://github.com/MongoEngine/flask-mongoengine/issues/247#issuecomment-243353219), I think we all agree that it's better to pass the most parameters as transparently as we can to avoid messing up with them or having to fix our code every time parameters are changed.

**Comment by jcass77** [comment]
Just chiming in here, but this issue also makes it difficult to use hosting services like [mLab](https://mlab.com). Not all of the subscription options include the default ``admin`` database, which means that one needs to be able to specify ``authentication_source`` (this was also raised in #256).

**Comment by fabian-rump** [comment]
@lafrech Currently passing tz_aware as a URL option (e.g. `mongodb://host:port/db?tz_aware=true`) does not have the expected effect, since pymongo always takes this option from the keyword arguments provided to the MongoClient (see [here](https://github.com/mongodb/mongo-python-driver/blob/master/pymongo/mongo_client.py#L357) and [here](https://github.com/mongodb/mongo-python-driver/blob/master/pymongo/mongo_client.py#L362)). I don't think that they will ever change this, since tz_aware is a Python specific thing and not related to the mongodb connection. So, is there any different solution?

**Comment by fabian-rump** [comment]
If anyone else is having this problem, here is the very dirty hack if have put in my application startup code until this gets fixed:
```
# Dirty monkey patching of MongoClient to enable tz_aware
_original_init = MongoClient.__init__


def _patched_init(*args, **kwargs):
    _original_init(*args, **kwargs, tz_aware=True)

MongoClient.__init__ = _patched_init
```

**Comment by lafrech** [comment]
Contributors willing to fix the connexion regression in 0.8 are welcome to join in https://github.com/MongoEngine/flask-mongoengine/issues/71#issuecomment-252554167.

**Comment by wojcikstefan** [comment]
Should be fixed in v0.8.1, v0.8.2, and v0.9.0.
